chore: migrate rollup-plugin-babel (#108)
rollup/plugins: packages/babel/README.md

[npm]: https://img.shields.io/npm/v/@rollup/plugin-babel
[npm-url]: https://www.npmjs.com/package/@rollup/plugin-babel
[size]: https://packagephobia.now.sh/badge?p=@rollup/plugin-babel
[size-url]: https://packagephobia.now.sh/result?p=@rollup/plugin-babel

[![npm][npm]][npm-url]
[![size][size]][size-url]
[![libera manifesto](https://img.shields.io/badge/libera-manifesto-lightgrey.svg)](https://liberamanifesto.com)

# @rollup/plugin-babel

> 🍣 Seamless integration between Rollup and Babel.

Reviewer: ```suggestion
🍣 A Rollup plugin for seamless integration between Rollup and Babel.
```

> ## Why?

Reviewer: @shellscape Should this section be moved down?

Author: I was thinking where this should be put when I was tweaking the README and I've decided to keep it here, at the top, as it made the most sense for me. Can, of course, change this but this acts kinda as an introduction to the whole thing, so it made sense to keep it at the beginning

Reviewer 2: I would say in 99% of cases, this is "meta" and should go after the really important bits (reqs, install, use, options). but the babel plugin does require some "do you really need this" knowledge up front. I'm good leaving this here as an exception.

- run the code through Rollup first, and _then_ pass it to Babel.

Both approaches have disadvantages – in the first case, on top of the additional configuration complexity, you may end up with Babel's helpers (like `classCallCheck`) repeated throughout your code (once for each module where the helpers are used). In the second case, transpiling is likely to be slower, because transpiling a large bundle is much more work for Babel than transpiling a set of small files.

Either way, you have to worry about a place to put the intermediate files, and getting sourcemaps to behave becomes a royal pain.

Using Rollup with `@rollup/plugin-babel` makes the process far easier.

## Requirements

This plugin requires an [LTS](https://github.com/nodejs/Release) Node version (v8.0.0+) and Rollup v1.20.0+.

## Install

```bash
npm install @rollup/plugin-babel --save-dev
```

## Usage

Create a `rollup.config.js` [configuration file](https://www.rollupjs.org/guide/en/#configuration-files) and import the plugin:

```js
import babel from '@rollup/plugin-babel';
import pkg from './package.json';

const config = {
  input: 'src/index.js',
  output: [
    {

>       file: pkg.module,

Reviewer: Let's make this example simpler and just hardcode `dir: 'output'` like the other READMEs.

Author: I'm not sure if people expect to use `dir` here - isn't it targeting multi-entry scenarios (so a more complex one)? I often find people wanting to have control over the single output filename.

Reviewer 2: @Andarist the real point of this section is an easy, frictionless intro into using the plugin. more complex, nuanced examples are typically located further down in either options sections or in "meta" sections after options that explain various concepts in more detail. this is a core tenant we've been following.

      format: 'esm'
    }
  ],

>   plugins: [babel({ babelHelpers: 'bundled' })]

Reviewer: ```suggestion
  plugins: [babel()]
```
Since this is the default its fine to omit and be consistent with other usage examples.

Author: This is default but relying on it results currently causes a console warn. The idea is that everyone should decide explicitly which one they want to use to improve the discoverability of this option.

};

export default config;
```

Then call `rollup` either via the [CLI](https://www.rollupjs.org/guide/en/#command-line-reference) or the [API](https://www.rollupjs.org/guide/en/#javascript-api).

## Options

This plugin respects Babel [configuration files](https://babeljs.io/docs/en/configuration) by default and they are generally the best place to put your configuration.

You can also run Babel on the generated chunks instead of the input files. Even though this is slower, it is the only way to transpile Rollup's auto-generated wrapper code to lower compatibility targets than ES5 (or most likely ES6 when Rollup 2 will be released), see [Running Babel on the generated code](#running-babel-on-the-generated-code) for details.

All options are as per the [Babel documentation](https://babeljs.io/docs/en/options), plus the following:

### `exclude`

Type: `String | RegExp | Array[...String|RegExp]`<br>

> A [minimatch pattern](https://github.com/isaacs/minimatch), or array of patterns, which specifies the files in the build the plugin should _ignore_. When relaying on Babel configuration files you can only exclude additional files with this option, you cannot override what you have configured for Babel itself.

Reviewer: relying*

### `include`

Type: `String | RegExp | Array[...String|RegExp]`<br>

> A [minimatch pattern](https://github.com/isaacs/minimatch), or array of patterns, which specifies the files in the build the plugin should _ignore_. When relaying on Babel configuration files you cannot include files already excluded there.

Reviewer: ```suggestion
A [minimatch pattern](https://github.com/isaacs/minimatch), or array of patterns, which specifies the files in the build the plugin should operate on. When relying on Babel configuration files you cannot include files already excluded there.
```
